prost-build/src/lib.rs

```diff
@@ -173,6 +173,20 @@ pub trait ServiceGenerator {
     /// Generates a Rust interface or implementation for a service, writing the
     /// result to `buf`.
     fn generate(&self, service: Service, buf: &mut String);
+    /// Finalizes the generation process.
+    ///
+    /// In case there's something that needs to be output at the end of the generation process, it
+    /// goes here. Similar to [`generate`](#method.generate), the output should be appended to
+    /// `buf.
+    ///
+    /// An example can be a module or other thing that needs to appear just once, not for each
+    /// service generated.
+    ///
+    /// This still can be called multiple times in a lifetime of the service generator, because it
+    /// can be used through multiple `.proto` files.
```

Comment:
I think it would be clearer as ', because it is called once per `.proto` file.'

Reply:
Oh I suppose this is only true if it's changed to run whether or not there are any services defined.